state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs

```diff
@@ -611,39 +651,204 @@ impl AptosDataClientInterface for MockAptosDataClient {
+
+        // Attempt to fetch and serve the new data
+        if self.data_beyond_highest_advertised && known_version < MAX_REAL_TRANSACTION_OUTPUT {
+            // Create a mock data client without timeout verification (to handle the internal requests)
+            let aptos_data_client = self.clone_without_timeout_verification();
+
+            // Determine the target ledger info
+            let known_epoch = self.find_known_epoch_for_version(known_version);
+            let target_ledger_info =
+                determine_targer_ledger_info(known_epoch, request_timeout_ms, &aptos_data_client)
+                    .await;
+
+            // Fetch the new transaction outputs
+            let outputs_with_proof = aptos_data_client
+                .get_transaction_outputs_with_proof(
+                    known_version + 1,
+                    known_version + 1,
+                    known_version + 1,
+                    self.aptos_data_client_config.response_timeout_ms,
+                )
+                .await
+                .unwrap()
+                .payload;
+
+            // Create and send a data client response
+            return Ok(create_data_client_response((
+                outputs_with_proof,
+                target_ledger_info,
+            )));
+        }
+
+        // Otherwise, emulate a network request timeout
+        Err(self.emulate_network_request_timeout())
     }
 
     async fn subscribe_to_transactions_with_proof(
         &self,
-        _subscription_request_metadata: SubscriptionRequestMetadata,
-        _include_events: bool,
-        _request_timeout_ms: u64,
+        subscription_request_metadata: SubscriptionRequestMetadata,
+        include_events: bool,
+        request_timeout_ms: u64,
     ) -> aptos_data_client::error::Result<
         Response<(TransactionListWithProof, LedgerInfoWithSignatures)>,
     > {
-        unimplemented!("TODO: Support me when we add unit tests!")
+        // Extract the known version, known epoch and the subscription stream index
+        let known_version_at_stream_start =
+            subscription_request_metadata.known_version_at_stream_start;
+        let known_epoch_at_stream_start = subscription_request_metadata.known_epoch_at_stream_start;
+        let subscription_stream_index = subscription_request_metadata.subscription_stream_index;
+
+        // Verify the request timeout
+        let data_request =
+            DataRequest::SubscribeTransactionsWithProof(SubscribeTransactionsWithProofRequest {
+                subscription_stream_metadata: SubscriptionStreamMetadata {
+                    known_version_at_stream_start,
+                    known_epoch_at_stream_start,
+                    subscription_stream_id: subscription_request_metadata.subscription_stream_id,
+                },
+                subscription_stream_index,
+                include_events,
+            });
+        self.verify_request_timeout(request_timeout_ms, false, true, data_request);
```

Comment:
nit: would it be clearer as `verify_request_timeout_value`? At first glance I thought a timeout was expected to be triggered.

Reply:
Yeah, makes sense!